update the description of the release process (#616)
pfnet-research/pfhedge: .github/CONTRIBUTING.md

make format
make check
```

> ## Release Procedures For Maintainers

Reviewer: ```suggestion
## Release Process for Maintainers
```

1. make branch for release whose name starts from "release/"
2. Version update pull request is automatically generated. Then, check and accept the pull request.

> 3. Create pull request from "release/*" to "main" and write release note on the body of the pull request.

Reviewer: ここ日本語で説明してもらえますか？生成されるPRと作るPRはそれぞれどういう役割？最初のPRはrelease/が何にマージされるPRですか？

> 1. make branch for release whose name starts from "release/"
> 2. Version update pull request is automatically generated. Then, check and accept the pull request.
> 3. Create pull request from "release/*" to "main" and write release note on the body of the pull request.

Reviewer: ```suggestion
1. Create a release branch named with the prefix "release/".
2. A pull request to update the version information in the release branch is automatically generated. Then, review and merge the pull request.
3. Create a pull request from "release/*" to "main", and write release notes in the description of the pull request.
```

> 4. After the release pull request was merged to main branch, pypi release and release page is automatically uploaded.
>     - please do not remove release branch by yourself. Github action will delete it after the procedures are completed.
> 
> Note: dev/develop branch is no longer used for development. Please make pull request directly into main branch.

Reviewer: ```suggestion
4. When the pull request is merged into the main branch, a PyPI release and a release page will be automatically configured.
    - Please do not remove the release branch by yourself. Github Actions will automatically delete the release branch once the release process is done.

NOTE: dev/develop branch is no longer used for development. Please make a pull request directly into main branch.
```

Author: a release page -> a github release pageに後で変更します
